toonz/sources/toonz/vectorinspector.cpp

```diff
@@ -0,0 +1,678 @@
+
+    if (vectorImage) {
+      QObject::disconnect(m_vectorImage, &TVectorImage::changedStrokes, this,
+                          &VectorInspectorPanel::onChangedStrokes);
+
+      QObject::connect(m_vectorImage, &TVectorImage::changedStrokes, this,
+                       &VectorInspectorPanel::onChangedStrokes);
+
+      QObject::disconnect(m_vectorImage, &TVectorImage::enteredGroup, this,
+                          &VectorInspectorPanel::onEnteredGroup);
+
+      QObject::connect(m_vectorImage, &TVectorImage::enteredGroup, this,
+                       &VectorInspectorPanel::onEnteredGroup);
+
+      QObject::disconnect(m_vectorImage, &TVectorImage::exitedGroup, this,
+                          &VectorInspectorPanel::onExitedGroup);
+
+      QObject::connect(m_vectorImage, &TVectorImage::exitedGroup, this,
+                       &VectorInspectorPanel::onExitedGroup);
+
+      QObject::disconnect(m_vectorImage, &TVectorImage::changedStrokeOrder,
+                          this, &VectorInspectorPanel::onStrokeOrderChanged);
+
+      QObject::connect(m_vectorImage, &TVectorImage::changedStrokeOrder, this,
+                       &VectorInspectorPanel::onStrokeOrderChanged);
+
+      QObject::disconnect(m_vectorImage, &TVectorImage::selectedAllStrokes,
+                          this, &VectorInspectorPanel::onSelectedAllStrokes);
+
+      QObject::connect(m_vectorImage, &TVectorImage::selectedAllStrokes, this,
+                       &VectorInspectorPanel::onSelectedAllStrokes);
+
+      QStandardItemModel *model =
+          m_vectorImage->getStrokeListData(parentWidget());
+      setSourceModel(model);
+      proxyModel->invalidate();
+
+      // Autosizing columns after the model is populated
+      for (int column = 1; column < model->columnCount(); ++column) {
+        proxyView->resizeColumnToContents(column);
+      }
+      proxyView->setColumnWidth(0, 30);  // Stroke column width
+    }
+
+  } else {
+    // disconnect all the signals which only apply when the current level is
+    // Vector
+    QObject::disconnect(TApp::instance()->getCurrentSelection(),
+                        &TSelectionHandle::selectionChanged, this,
+                        &VectorInspectorPanel::onSelectionChanged);
+
+    QObject::disconnect(TApp::instance()->getCurrentTool(),
+                        &ToolHandle::toolSwitched, this,
+                        &VectorInspectorPanel::onToolSwitched);
+
+    QObject::disconnect(TApp::instance()->getCurrentScene(),
+                        &TSceneHandle::sceneSwitched, this,
+                        &VectorInspectorPanel::onSceneChanged);
+
+    QObject::disconnect(TApp::instance()->getCurrentScene(),
+                        &TSceneHandle::sceneChanged, this,
+                        &VectorInspectorPanel::onSceneChanged);
+
+    setSourceModel(new MultiColumnSortProxyModel);
+  }
+
+  m_field->setText("Level type: " + QString::fromStdString(levelType) +
+                   " at: " + utcNow.toString(Qt::ISODate));
```

Comment:
Is there a particular reason for showing current time?  I don't think it provides any value.

As this is for vectors only, I don't particularly see the need for Level Type information.  If you are not on a Vector level, the panel information should be blank.  If anything should be displayed, perhaps the Level Name and Frame number might be of more value than Type and Time.

Reply:
Timestamp so I could tell if the panel was updating. If the timestamp was "frozen" at a time in the past I would recognize it was not updating. In a screen capture as part of discussion troubleshooting Vector fill issues for example, the timestamp could be there. Similarly, the level type is to show what the panel is currently pointing to.
I'll look at putting Level Name and Frame number instead.

Reply:
I can update the window title in a manner similar to other windows or panels have done.
Do you have a preference for separator?
I see several in use. Here I am using vertical bars, it looks clean. 
I see single or double colon used in other windows. 
![image](https://github.com/tahoma2d/tahoma2d/assets/44948041/4e5ff0d7-b0a6-4583-9d53-cd3748e40757)

Reply:
If you are going to put it in the title bar area, make sure the title bar doesn't hide if the pane is docked and locked otherwise the information is lost.  If I had a choice, follow the Scene Viewer separator.

Reply:
I trimmed things quite a bit getting the panel to not have extra "stuff" left over from its origin as a mash-up between HistoryPanel and an example from Qt. So, currently it is not dockable. I will bring that functionality back and make sure the title and info doesn't disappear when docked.

Reply:
Completed.